helix-term/src/ui/completion.rs

```diff
@@ -79,7 +92,39 @@ impl menu::Item for CompletionItem {
                 Some(lsp::CompletionItemKind::ENUM) => "enum",
                 Some(lsp::CompletionItemKind::KEYWORD) => "keyword",
                 Some(lsp::CompletionItemKind::SNIPPET) => "snippet",
-                Some(lsp::CompletionItemKind::COLOR) => "color",
+                Some(lsp::CompletionItemKind::COLOR) => {
+                    let maybe_hex_color = match &item.documentation {
+                        Some(Documentation::String(text)) => Some(text),
+                        Some(Documentation::MarkupContent(MarkupContent { value, .. })) => {
+                            Some(value)
+                        }
+                        None => None,
+                    };
```

Comment:
```suggestion
                let maybe_hex_color = item.documentation.map(|doc| match doc {
                    Documentation::String(text) => text,
                    Documentation::MarkupContent(MarkupContent { value, .. }) => value,
                });

```

Reply:
I think it's better but I tried and looks like the `map` will move out of the `item.documentation` so I would need to use `clone`, I left it as is. If you find an approach that doesn't use `clone` let me know

```rs
                    let maybe_hex_color = item.documentation.map(|doc| match doc {
                        Documentation::String(text) => text,
                        Documentation::MarkupContent(MarkupContent { value, .. }) => value,
                    });

                    maybe_hex_color.map_or(Span::raw("color"), |hex| {
                        let maybe_color = Color::from_hex(&hex);

                        match maybe_color {
                            Some(color) => Span::styled("■", Style::default().fg(color)),
                            None => Span::raw(""),
                        }
                    })
```

```
you can `clone` the value and consume it, but this might not be your desired behavior: `<std::option::Option<helix_lsp::helix_lsp_types::Documentation> as Clone>::clone(&`, `)`
cannot move out of `item.documentation` which is behind a shared reference
help: consider calling `.as_ref()` or `.as_mut()` to borrow the type's contents
cannot move out of `item.documentation` which is behind a shared reference
move occurs because `item.documentation` has type `std::option::Option<helix_lsp::helix_lsp_types::Documentation>`, which does not implement the `Copy` trait
consider cloning the value if the performance cost is acceptable: `.clone()`
`item.documentation` moved due to this method call
```